Could not resolve "../screen"

## Description 
Encountered issue when bundling with [esbuild](https://esbuild.github.io/getting-started/):

```
Error: Build failed with 1 error:
../../../common/temp/node_modules/.pnpm/@oclif+core@1.0.2/node_modules/@oclif/core/lib/errors/errors/cli.js:40:38: error: Could not resolve "../screen"
  ...20 lines omitted...
        suggestion: ''
      },
      notes: [],
      pluginName: '',
      text: 'Could not resolve "../screen"'
    }
  ],
  warnings: []
}
ERROR: "build:esbuild" exited with 1.
```

## Cause

Is there a reason you guys are mixing inline `requires` with `import` syntax? I've found this can cause a lot of issues in the past and can't see any obvious benefit?

#### Examples

- https://github.com/oclif/core/blob/main/src/errors/errors/cli.ts#L36
- https://github.com/oclif/core/blob/main/src/errors/errors/cli.ts#L49-L50

- https://github.com/oclif/core/blob/main/src/errors/errors/cli.ts#L54

The last link  is specifically the line that throws the error described, but I imagine there will be a lot of others once this is resolved if this one has managed to get through your CI / tests given that the `../screen` file doesn't exist.

See here for `../` directory (note the `screen` file exists in the next level up):
- https://github.com/oclif/core/tree/main/src/errors

This would be avoided by enforcing [typescript-eslint/no-require-imports](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-require-imports.md) with a hard `error` during build. 

These inline requires are not handled well by the `tsc` compiler whereas using top of page `imports` will have definitely failed to build, alerting you to the error much sooner before publishing.


## Context
- @oclif/core 1.0.2
- esbuild 0.13.9
- typescript 4.4.3

--- a/src/errors/errors/cli.ts
+++ b/src/errors/errors/cli.ts
@@ -1,10 +1,9 @@
-// tslint:disable no-implicit-dependencies
+import * as chalk from 'chalk'
+import * as indent from 'indent-string'
+import * as cs from 'clean-stack'
+import * as wrap from 'wrap-ansi'
 
-import * as Chalk from 'chalk'
-import Clean = require('clean-stack')
-import Indent = require('indent-string')
-import * as Wrap from 'wrap-ansi'
-
+import * as screen from '../../screen'
 import {config} from '../config'
 import {PrettyPrintableError, OclifError} from '../../interfaces/errors'
 
@@ -33,8 +32,7 @@ export class CLIError extends Error implements OclifError {
   }
 
   get stack(): string {
-    const clean: typeof Clean = require('clean-stack')
-    return clean(super.stack!, {pretty: true})
+    return cs(super.stack!, {pretty: true})
   }
 
   /**
@@ -46,12 +44,8 @@ export class CLIError extends Error implements OclifError {
       return this.stack
     }
 
-    const wrap: typeof Wrap = require('wrap-ansi')
-    const indent: typeof Indent = require('indent-string')
-
     let output = `${this.name}: ${this.message}`
-    // eslint-disable-next-line node/no-missing-require
-    output = wrap(output, require('../screen').errtermwidth - 6, {trim: false, hard: true} as any)
+    output = wrap(output, screen.errtermwidth - 6, {trim: false, hard: true} as any)
     output = indent(output, 3)
     output = indent(output, 1, {indent: this.bang, includeEmptyLines: true} as any)
     output = indent(output, 1)
@@ -59,12 +53,9 @@ export class CLIError extends Error implements OclifError {
   }
 
   get bang() {
-    let red: typeof Chalk.red = ((s: string) => s) as any
     try {
-      red = require('chalk').red
+      return chalk.red(process.platform === 'win32' ? '»' : '›')
     } catch {}
-
-    return red(process.platform === 'win32' ? '»' : '›')
   }
 }
 
@@ -76,12 +67,9 @@ export namespace CLIError {
     }
 
     get bang() {
-      let yellow: typeof Chalk.yellow = ((s: string) => s) as any
       try {
-        yellow = require('chalk').yellow
+        return chalk.yellow(process.platform === 'win32' ? '»' : '›')
       } catch {}
-
-      return yellow(process.platform === 'win32' ? '»' : '›')
     }
   }
 }

